test/dotnet-watch.FunctionalTests/ProjectToolScenario.cs

```diff
@@ -0,0 +1,137 @@
+        {
+            var destinationFolder = Path.Combine(WorkFolder, Path.GetFileName(projectFolder));
+            Console.WriteLine($"Copying project {projectFolder} to {destinationFolder}");
+
+            Directory.CreateDirectory(destinationFolder);
+
+            foreach (string directory in Directory.GetDirectories(projectFolder, "*", SearchOption.AllDirectories))
+            {
+                Directory.CreateDirectory(directory.Replace(projectFolder, destinationFolder));
+            }
+
+            foreach (string file in Directory.GetFiles(projectFolder, "*.*", SearchOption.AllDirectories))
+            {
+                File.Copy(file, file.Replace(projectFolder, destinationFolder), true);
+            }
+        }
+
+        public void AddNugetFeed(string feedName, string feed)
+        {
+            string tempNugetConfigFile = Path.Combine(WorkFolder, NugetConfigFileName);
+
+            var nugetConfig = XDocument.Load(tempNugetConfigFile);
+            var packageSource = nugetConfig.Element("configuration").Element("packageSources");
+            packageSource.Add(new XElement("add", new XAttribute("key", feedName), new XAttribute("value", feed)));
+            using (var stream = File.OpenWrite(tempNugetConfigFile))
+            {
+                nugetConfig.Save(stream);
+            }
+        }
+
+        public void AddToolToProject(string projectName, string toolName)
+        {
+            var projectFile = Path.Combine(WorkFolder, projectName, "project.json");
+            Console.WriteLine($"Adding {toolName} to {projectFile}");
+
+            var projectJson = JObject.Parse(File.ReadAllText(projectFile));
+            projectJson.Add("tools", new JObject(new JProperty(toolName, "1.0.0-*")));
+            File.WriteAllText(projectFile, projectJson.ToString());
+        }
+
+        public void Restore(string project = default(string))
+        {
+            if (project == default(string))
+            {
+                project = WorkFolder;
+            }
+            else
+            {
+                project = Path.Combine(WorkFolder, project);
+            }
+
+            var restore = ExecuteDotnet($"restore", project);
+            restore.WaitForExit();
+
+            if (restore.ExitCode != 0)
+            {
+                throw new Exception($"Exit code {restore.ExitCode}");
+            }
+        }
+
+        private void CreateTestDirectory()
+        {
+            Directory.CreateDirectory(WorkFolder);
+            var nugetConfigFilePath = FindNugetConfig();
+
+            string tempNugetConfigFile = Path.Combine(WorkFolder, NugetConfigFileName);
+            File.Copy(nugetConfigFilePath, tempNugetConfigFile);
+        }
+
+        public Process ExecuteDotnet(string arguments, string workDir)
+        {
+            Console.WriteLine($"Running dotnet {arguments} in {workDir}");
+
+            var psi = new ProcessStartInfo("dotnet", arguments)
```

Comment:
Curious if this adds a lot of spew in the tests. If so, we might want to suppress output when successful to make test output cleaner.